src/components/base/b-virtual-scroll/interface/component.ts

```diff
@@ -0,0 +1,282 @@
+/*!
+ * V4Fire Client Core
+ * https://github.com/V4Fire/Client
+ *
+ * Released under the MIT license
+ * https://github.com/V4Fire/Client/blob/master/LICENSE
+ */
+
+import type bVirtualScroll from 'components/base/b-virtual-scroll/b-virtual-scroll';
+
+/**
+ * Component state.
+ *
+ * @typeParam DATA - Instance of the data element.
+ * @typeParam RAW_DATA - The data loaded from the server but not yet processed.
+ * This type parameter determines the type of the {@link VirtualScrollState.lastLoadedRawData} property
+ */
+export interface VirtualScrollState<DATA = object, RAW_DATA = unknown> {
+	/**
+	 * The largest component index of type `item` that appeared in the viewport.
+	 */
+	maxViewedItem: CanUndef<number>;
+
+	/**
+	 * The largest component index of any type that appeared in the viewport.
+	 */
+	maxViewedChild: CanUndef<number>;
+
+	/**
+	 * The number of components of type `item` that have not yet been visible to the user.
+	 */
+	itemsTillEnd: CanUndef<number>;
+
+	/**
+	 * The number of components of any type that have not yet been visible to the user.
+	 */
+	childTillEnd: CanUndef<number>;
+
+	/**
+	 * The current page number for loading data.
+	 * It changes after each successful data load.
+	 */
+	loadPage: number;
+
+	/**
+	 * The current page number for rendering data.
+	 * It changes after each successful rendering.
+	 */
+	renderPage: number;
+
+	/**
+	 * Indicates if the last loaded data is empty.
+	 */
+	isLastEmpty: boolean;
+
+	/**
+	 * Indicates if the last data load ended with an error.
+	 */
+	isLastErrored: boolean;
+
+	/**
+	 * Indicates if the component is in the initial loading state.
+	 */
+	isInitialLoading: boolean;
+
+	/**
+	 * Indicates if the component is in the initial rendering state.
+	 */
+	isInitialRender: boolean;
+
+	/**
+	 * Indicates if the component has stopped making requests.
+	 */
+	isRequestsStopped: boolean;
```

Comment:
Так как requests во множественном числе, то надо использовать `are`.

```suggestion
	areRequestsStopped: boolean;
```

Однако, здесь и выше есть несогласованность времени: `stopped` - в прошедшем времени, а используется `is`, который относится к настоящему, правильнее было бы так:

```suggestion
	haveRequestsStopped: boolean;
```

`Is` или `are` в данном случае можно использовать, если бы была такая формулировка:

```
Are requests being stopped?
```

То есть: "останавливаются ли в данный момент запросы"